docs/tx-malleability.md

```diff
@@ -0,0 +1,68 @@
+* transactions could potentially get executed more than once (faulty replay protection)
+
+If a client generates a transaction, keeps a record of its hash and then attempts to query nodes to check the transaction's status, this process may falsely conclude that the transaction had not been processed if an intermediary
+processor decoded and re-encoded the transaction with different encoding rules (either maliciously or unintentionally).
+As long as no malleability is present the signature bytes themselves, clients SHOULD query transactions by signature instead of hash.
+
+Not being cognizant of this risk may lead clients to submit the same transaction multiple times if they believe that 
+earlier transactions had failed or gotten lost in processing, which is somewhat akin to faulty replay protection.
+If the state machine were to rely on transaction hashes as a replay mechanism itself, this would be faulty and not 
+provide the intended replay protection. Instead, the state machine should rely on deterministic representations of
+transactions rather than raw encoding if they want to provide some replay protection that doesn't rely on a monotonically
+increasing account sequence number.
+
+
+## Sources of Malleability
+
+### Non-deterministic Protobuf Encoding
+
+Cosmos SDK transactions are encoded using protobuf binary encoding when they are submitted to the network. Protobuf binary is not inherently a deterministic encoding meaning that the same logical payload could have several valid bytes representations. In a basic sense, this means that protobuf in general can be decoded and re-encoded to produce a different byte stream (and thus different hash) without changing the logical meaning of the bytes. [ADR 027: Deterministic Protobuf Serialization](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-027-deterministic-protobuf-serialization.md) describes in detail what needs to be done to produce what we consider to be a "canonical", deterministic protobuf serialization. Briefly, the following sources of malleability at the encoding level have been identified and are addressed by this specification:
+* fields can be emitted in any order
+* default field values can be included or omitted, and this doesn't change meaning unless `optional` is used
+* `repeated` fields of scalars may use packed or "regular" encoding
+* `varint`s can include extra ignored bits
+* extra fields may be added and are usually simply ignored by decoders. [ADR 020](https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-020-protobuf-transaction-encoding.md#unknown-field-filtering) specifies that in general such extra fields should cause messages and transactions to be rejected)
+
+When using `SIGN_MODE_DIRECT` most of the above malleabilities will not be tolerated because:
```

Comment:
here we say "most" - does that mean some malleabilities will be tolerated? if so, what are they?

Reply:
I guess it's fair to say that none of these malleabilities will be tolerated. Will update. The only tolerated malleability would be malleability in the signature bytes themselves.